stl/inc/format

```diff
@@ -1985,6 +2103,109 @@ _NODISCARD _OutputIt _Fmt_write(
     return _Fmt_write(_STD move(_Out), basic_string_view<_CharT>{_Value}, _Specs, _Locale);
 }
 
+inline constexpr char16_t _U16_width_estimate_bounds[] = { // Per N4885 [format.string.std]/11
+    0x1100u, 0x1160u, 0x2329u, 0x232Bu, 0x2E80u, 0x303Fu, 0x3040u, 0xA4D0u, 0xAC00u, 0xD7A4u, 0xF900u, 0xFB00u, 0xFE10u,
+    0xFE1Au, 0xFE30u, 0xFE70u, 0xFF00u, 0xFF61u, 0xFFE0u, 0xFFE7u};
+
+inline constexpr char32_t _U20_width_estimate_bounds[] = { // Per N4885 [format.string.std]/11
+    0x1F300u, 0x1F650u, 0x1F900u, 0x1FA00u, 0x20000u, 0x2FFFEu, 0x30000u, 0x3FFFEu};
+
+template <auto& _Bounds>
+_NODISCARD constexpr int _Unicode_width_estimate(const char32_t _Ch) noexcept {
+    // Computes the width estimation for unicode characters from N4885 [format.string.std]/11
+    int _Result = 1;
+    for (auto const& _Bound : _Bounds) {
+        if (_Ch < _Bound) {
+            return _Result;
+        }
+        _Result ^= 1;
+    }
+    return 1;
+}
+
+_NODISCARD inline int _Estimate_character_width(const char* _Ptr, const int _Units, const _Cvtvec& _Cvt) {
+    // Return an estimate for the width of the character composed of _Units code units,
+    // whose first code unit is denoted by _Ptr.
+    auto _Ch = static_cast<char32_t>(*_Ptr);
+    if (_Cvt._Mbcurmax != 4) {
+        // not a Unicode encoding; estimate width as 1
+        return 1;

```

Comment:
```suggestion
        // not a Unicode encoding; estimate width as the number of code units
        return _Units;
```

This would match the existing practice that double-byte code points in legacy east Asian encodings are displayed twice as wide.

Reply:
Good catch! I wasn't sure about this and had intended to get feedback on using `_Units` instead in code review. Changing.